### PR TITLE
upcxx: remove warnings from package

### LINKED ISSUE #12800: upcxx: CROSS warning pollutes all invocations of Spack
Since the addition of `upcxx`, **all** invocations of Spack on systems without `aprun` result in a warning message:
```console
==> Warning: CROSS has not been set, however cannot detect scheduler.
```
This is somewhat sub-optimal to say the least, and results from the invocation of `detect_scheduler()` in order to provide the default of the `scheduler` variant, which is executed whenever the recipes are parsed.

I would suggest that an alternative scheme be found to set the default and verify that the setting is valid for the current system without producing a warning on a large class of non-HPC systems.

### PR DESCRIPTION
**Pull request by tgamblin**
fixes #12800 

Warnings in `detect_scheduler()` print out whenever the package is loaded, which  means you can see them in Spack commands that load all packages but have nothing to do with UPC++ (e.g., `spack clean -a`,  `spack dependents`, and others).

- [x] Remove these and other warnings, since the scheduler is visible in the package variant anyway.

- [x] Also remove the `user` scheduler option from `upcxx`, as it depends on the user's environment in a way that is not likely to be the same across systems.

@bonachea, @mjacquel can you take a look at this one?  The main thing I need feedback on is what the user setting for `CROSS` can be.  Is it needed?  Can it be something other than SLURM or ALPS?  If not we should remove it.  If so, then it should be changed to be a string-valued variant, so the user can set its value in a way Spack will actually understand.   Basically we're trying to avoid environment-dependent things in Spack --  one hash should map to one build.

Just for my own edification, does the UPC++  build link to different libraries depending on what CROSS is set to?  Does it actually affect the final binary somehow, or is that just a build-time thing?

### CODE REVIEW

File: var/spack/repos/builtin/packages/upcxx/package.py [hunk 1]
```diff
@@ -9,26 +9,12 @@
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/upcxx/package.py]:
```
var/spack/repos/builtin/packages/upcxx/package.py:7: [F401] 'llnl.util.tty' imported but unused
var/spack/repos/builtin/packages/upcxx/package.py:8: [F401] 'os' imported but unused
```

**Comment by mjacquel** [comment]
I see that you have removed the support of the CROSS environment variable if it has already been set. I guess letting the user/admin chose its value by picking up a variant serves the same purpose.

Although I cannot officially approve this PR, I approve this PR anyway ;-)

**Comment by bonachea** [comment]
@tgamblin - `$CROSS` is the general means by which a cross-compiled GASNet target platform is selected at configure time. It encapsulates a complete set of build settings, some of which cannot be detected on the frontend node. CURRENTLY there is only one cross-compiled platform that is also supported by UPC++, the Cray XC, and the only dimension we cannot auto-detect on this platform is the scheduler so this very general decision has been reduced to the very specific option of ALPS vs. SLURM. You'll notice the actual values of CROSS are things like "cray-aries-slurm", those select a cross-configure script from [this directory](https://gasnet.lbl.gov/dist-ex/other/contrib/). The ALPS vs SLURM distinction in particular does not affect the ABI, just generated scripts used in job launching. However CROSS values that specify a different target platform (for example the BlueGene) obviously would affect ABI.

Removing CROSS detection/reporting from this Spack package is fine for today, but if/when we add another cross-compiled UPC++ platform it may have to return and/or be expanded (possibly through use of variants). Although I suppose as long as Spack doesn't actively clear/override this envvar, admins should still be able to pass it through (right?).

**Comment by bonachea** [comment]
This pull request has been superseded by pull request #12880 and should be closed

**Pull request closed without merging**